src/main/java/org/broadinstitute/hellbender/utils/clipping/ClippingOp.java

```diff
@@ -176,34 +176,44 @@ private GATKRead applyREVERT_SOFTCLIPPED_BASES(final GATKRead read) {
     }
 
     /**
-     * Given a cigar string, get the number of bases hard or soft clipped at the start
+     * Given two cigar strings corresponding to the a read that has been softcliped and before, returns an integer
+     * corresponding to the number of reference bases that the new string must be offset by in order to have the
+     * correct start according to the reference.
+     *
+     * @VisibleForTesting
      */
-    private int getNewAlignmentStartOffset(final Cigar __cigar, final Cigar __oldCigar) {
-        int num = 0;
-        for (final CigarElement e : __cigar.getCigarElements()) {
+    static int getNewAlignmentStartOffset(final Cigar clippedCigar, final Cigar oldCigar) {
+        int readBasesBeforeReference = 0;
+
+        int basesBeforeReferenceOld = 0;
```

Comment:
Add a one-line comment for each of these variables explaining what it represents